src/core/MixHelpers.cpp

```diff
@@ -90,55 +91,58 @@ void setNaNHandler( bool use )
 }
 
 /*! \brief Function for sanitizing a buffer of infs/nans - returns true if those are found */
-bool sanitize( sampleFrame * src, int frames )
+bool sanitize( SampleFrame * src, int frames )
 {
 	if( !useNaNHandler() )
 	{
 		return false;
 	}
 
 	bool found = false;
-	for( int f = 0; f < frames; ++f )
+	for (int f = 0; f < frames; ++f)
 	{
-		for( int c = 0; c < 2; ++c )
+		auto& currentFrame = src[f];
+
+		if (currentFrame.containsInf() || currentFrame.containsNaN())
 		{
-			if( std::isinf( src[f][c] ) || std::isnan( src[f][c] ) )
-			{
-				#ifdef LMMS_DEBUG
+			#ifdef LMMS_DEBUG
 					// TODO don't use printf here
 					printf("Bad data, clearing buffer. frame: ");
-					printf("%d: value %f\n", f, src[f][c]);
-				#endif
-				for( int f = 0; f < frames; ++f )
-				{
-					for( int c = 0; c < 2; ++c )
-					{
-						src[f][c] = 0.0f;
-					}
-				}
-				found = true;
-				return found;
-			}
-			else
-			{
-				src[f][c] = std::clamp(src[f][c], -1000.0f, 1000.0f);
-			}
+					printf("%d: value %f, %f\n", f, currentFrame.left(), currentFrame.right());
+			#endif
+			found = true;
+			break;
+		}
+		else
+		{
+			// TODO What is this supposed to do? This corresponds to saying: "Pushing a signal with 60 dbFS is no problem".
+			currentFrame.clamp(sample_t(-1000.0), sample_t(1000.0));
```

Comment:
I have removed the comment with commit ae919ec1 and have simplified the `sanitize` function a bit with commit 0fee3b6c.

Does the clamping really need to be done for every call to `sanitize` though? I think it gets called a lot and I wonder if it should not suffice to have a final hard limiter on the master. This could be a plugin that's on the master by default and that could be adjusted with regards to the maximum dbFS value.